loader/src/sources/cdc/api.js

```diff
@@ -0,0 +1,282 @@
+        yield result;
+      }
+
+      if (results.length < PAGE_SIZE) {
+        return;
+      } else {
+        offset += PAGE_SIZE;
+      }
+    } catch (e) {
+      error(`Error fetching CDC data`, e);
+      return;
+    }
+  }
+}
+
+function formatStore(storeItems) {
+  if (storeItems.length == 0) {
+    return null;
+  }
+
+  let result;
+  Sentry.withScope((scope) => {
+    const base = storeItems[0];
+    scope.setContext("location", {
+      id: base.provider_location_guid,
+      name: base.loc_name,
+      provider: "cdc",
+    });
+
+    const storeExternalId = getStoreExternalId(base);
+    if (!storeExternalId) {
+      return null;
+    }
+
+    const addressLines = [
+      base.loc_admin_street1,
+      base.loc_admin_street2,
+    ].filter(Boolean);
+
+    const metaFields = [
+      "insurance_accepted",
+      "walkins_accepted",
+
+      // TODO: better structure open hours information
+      "sunday_hours",
+      "monday_hours",
+      "tuesday_hours",
+      "wednesday_hours",
+      "thursday_hours",
+      "friday_hours",
+      "saturday_hours",
+    ];
+    const meta = {};
+    for (const field of metaFields) {
+      meta[field] = base[field];
+    }
+
+    const productFields = [
+      "ndc",
+      "med_name",
+      "in_stock",
+      "supply_level",
+      "quantity_last_updated",
+    ];
+    const productList = storeItems.map((item) => {
+      const out = {};
+      for (const field of productFields) {
+        out[field] = item[field];
+      }
+      return out;
+    });
+
+    result = {
+      id: `cdc:${base.provider_location_guid}`,
+      external_ids: [
+        ["vaccines_gov", base.provider_location_guid],
+        storeExternalId,
+      ],
+      name: titleCase(base.loc_name),
+      provider: "cdc",
+
+      address_lines: addressLines,
+      city: base.loc_admin_city,
+      state: base.loc_admin_state,
+      postal_code: base.loc_admin_zip,
+      position: {
+        longitude: base.longitude,
+        latitude: base.latitude,
+      },
+      info_phone: base.loc_phone,
+      info_url: base.web_address,
+      meta,
+
+      availability: {
+        source: "cdc",
+        checked_at: new Date().toISOString(),
+        valid_at: formatValidAt(productList),
+        available_count: getAvailableCount(productList),
+        available: formatAvailable(productList),
+        capacity: formatCapacity(productList),
+        products: formatProductTypes(productList),
+      },
+    };
+  });
+  return result;
+}
+
+const systemNameRe = {
+  costco: /^Costco/i,
+  cvs: /^CVS/i,
+  kroger: /^Kroger/i,
+  publix: /^Publix/i,
+  rite_aid: /^Rite Aid/i,
+  safeway: /^SAFEWAY/i,
+  sams_club: /^Sams Club/i,
+  walgreens: /^Walgreens/i,
+  walmart: /^Walmart/i,
+};
+
+function getStoreExternalId(store) {
+  const m = store.loc_store_no.match(/^[A-Z]*(\d+)$/); // handle cases like RA105587 -> 105587
+  if (!m) {
+    return null;
+  }
+  const storeNumber = m[1];
```

Comment:
Grrrr, looking at this again now, it’s not working well in the case where these values are VTrckS PINs instead of store numbers. (I didn’t even realize that was the case for some values in this column before!)

For VTrckS, the first 3 characters are an identifier for who issued the PIN, so in the comment above, `RA1` designates that this was issued by Rite Aid (the major pharmacies get to issue their own pins, then there are ones issued by CDC [not sure on the prefix there] and by state departments of health, e.g. NJ’s is `NJ<number>`). I have no idea where any official documentation for that is; I kept trying to find it but ultimately only gleaned this info through several conversations with CDC and DoH folks.

So anyway, the Rite Aid store number we should wind up with when handed `RA105587` is actually `5587` (so we also need to remove the leading zeroes to make it as normal as possible). (See our existing record for this example: https://getmyvax.org/api/edge/locations/rite_aid:5587)

Reply:
Just took a stab at handling things both ways, using the guidelines you shared in Slack

> The format so far as I understand it (may not be 100% correct):
> - Always (?) 8 characters.
> - First 3 characters designates who issued the PIN (major pharmacies, states, CDC directly, not sure if there are other issuers/assigners/whatever)
> - States are <2-letter postal abbreviation><digit>
> - Pharmacies are <2-letter abbreviation>1 or <3-letter abbreviation>. The remaining characters are usually the store number, but are really up to the pharmacy as the issuer.
> - Remaining characters are up to the issuer and could be anything.